src/server/scripts/Kalimdor/RuinsOfAhnQiraj/instance_ruins_of_ahnqiraj.cpp

```diff
@@ -84,9 +84,30 @@ class instance_ruins_of_ahnqiraj : public InstanceMapScript
         {
             if (GetBossState(DATA_KURINNAXX) == DONE && GetBossState(DATA_RAJAXX) != DONE)
             {
-                if (!_andorovGUID)
+                Map::PlayerList const &playerList = instance->GetPlayers();
+
+                if (playerList.IsEmpty())
+                    return;
+
+                bool anyPlayerInCombat = false;
+                for (Map::PlayerList::const_iterator itr = playerList.begin(); itr != playerList.end(); ++itr)
```

Comment:
You can try instance->DoForAllPlayers here which could return the boolean as well when player in combat is found

Reply:
I don't understand this part, why are we checking if any player is in combat? This doesn't look right

Reply:
Because if you don't, a raid could resurrect Andorov during combat by having someone disconnect and log back in, triggering this statement. Too many Andorovs!

In my rewrite I'm hoping to create a resummon methodology that doesn't rely on this, but for now it fixes the current implementation without making it cheese-able.

Reply:
Yes, check if the encounter is in progress

Reply:
I'll look again and see if I can see how to do that. I didn't think there was an "In Progress" status for the event, but maybe I missed it.

Reply:
There might not be one, in that case, check if _rajaxxWaveCounter > 0